#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PKG = "jsk_robot_startup"

gen = ParameterGenerator()

#       name    type     level     description     default      min      max
gen.add("periodic", bool_t, 0, "Publish periodically", False)
gen.add("periodic_rate", double_t, 0, "Publish rate. Valid only if ~periodic is enabled", 1.0, 0.1, 100)
gen.add("threshold", double_t, 0, "Threshold to determine if joints are moved", 0.01, 0.001, 1.0)
gen.add("rotation_factor", double_t, 0, "Factor for rotational distance", 0.1, 0.001, 1.0)

exit(gen.generate(PKG, PKG, "JointStatesThrottle"))
